xarray/plot/utils.py

```diff
@@ -221,8 +221,14 @@ def _infer_xy_labels(darray, x, y):
         if darray.ndim != 2:
             raise ValueError('DataArray must be 2d')
         y, x = darray.dims
-    elif x is None or y is None:
-        raise ValueError('cannot supply only one of x and y')
+    elif x is None:
+        if y not in darray.dims:
+            raise ValueError('y must be a coordinate variable')
```

Comment:
@fmaussion : You are proposing the same message as @shoyer above. My comment seems still valid to me. 

Why focus on 'if x is not supplied'? Why no equivalent 'if y is not supplied'? I feel like I'm missing something.